s3_management/manage.py

```diff
@@ -212,6 +213,23 @@ def normalize_package_version(self: S3IndexType, obj: str) -> str:
     def obj_to_package_name(self, obj: str) -> str:
         return path.basename(obj).split('-', 1)[0]
 
+    def fetch_checksum_from_s3(self, s3_key):
+        s3_key = s3_key.replace("%2B", "+")
+        try:
+            response = CLIENT.get_object_attributes(
+                Bucket=BUCKET,
+                Key=s3_key,
+                ObjectAttributes=['Checksum']
+            )
+            checksum = response['Checksum']['ChecksumSHA256']
+            return checksum
+        except NoCredentialsError:
+            print("No AWS credentials found")
+            return None
```

Comment:
Why not just let this propogate? Otherwise it'll just repeatedly print the error and generate a sub-optimal index.

Reply:
My thinking was that the index page should still generate as it does now even if there is no hash available to provide.   I should definitely remove the print statements.  I am not sure what exceptions would be raised before the `Checksum` values get added to the objects.

Reply:
(Did this mean to get commented below? on https://github.com/pytorch/builder/pull/1433/files/b3799a8eb42b56118a4bc0658902875734888721#r1261472864? If not I'm not following the thread here. If the credentials aren't found we really can't know if there's a checksum or not)

Reply:
I mean its the same code block in my mind -- I am just saying I don't know what exceptions we want to catch and handle for the case of the checksum isn't available yet -- maybe we don't care about `NoCredentialsError` but I think we need to handle _something_

Reply:
I think it's safe to assume:
- The runner has valid credentials to the object
- The runner has the correct permissions to access the attributes
- S3 is alive and well and will return good JSON.

Because otherwise if these assumptions aren't held, there's a chance the script generates an `index.html` without the checksums and we're back to square 1.

So I think the only thing to handle here is "this object has no checksum attribute".

Which means the code looks roughly like:
```python
response = CLIENT.get_object_attributes(
    Bucket=BUCKET,
    Key=s3_key,
    ObjectAttributes=["Checksum"]
)
checksum = response.get("Checksum", {}).get("ChecksumSHA256", None)
```

without any `try`/`except`.

FWIW I tested your code against a work bucket and got `TypeError: expected string or bytes-like object`. Switching the argument for `Bucket` to the string bucket name then gives me a valid response.

So probably use `BUCKET.name` as well.

---

Here's some local testing...

```python
>>> import boto3
>>> BUCKET=S3.Bucket("<bucketname>")
>>> CLIENT = boto3.client("s3")
>>> response = CLIENT.get_object_attributes(Bucket=BUCKET.name,Key="<object-without-checksum>",ObjectAttributes=['Checksum'])
>>> list(response)
['ResponseMetadata', 'LastModified', 'VersionId']
>>> response.get("Checksum", {}).get("ChecksumSHA256", None)
>>> response = CLIENT.get_object_attributes(Bucket=BUCKET.name,Key="<object-WITH-checksum>",ObjectAttributes=['Checksum'])
>>> list(response)
['ResponseMetadata', 'LastModified', 'Checksum']
>>> response.get("Checksum", {}).get("ChecksumSHA256", None)
'7Uc3AtM1Dlc4XiQ8A6NkQdV9JMU4liPtz7wOd+RRCTE='
```